File: spec/cpp_stl/test_hello_world.cpp
#include <boost/test/unit_test.hpp>

#include <hello_world.h>

#include <iostream>
#include <fstream>

BOOST_AUTO_TEST_CASE(test_hello_world) {
//    hello_world* r = hello_world::from_file("src/fixed_struct.bin");

    std::ifstream ifs("src/fixed_struct.bin", std::ifstream::binary);
    kaitai::kstream ks(&ifs);
    hello_world_t* r = new hello_world_t(&ks);

    BOOST_CHECK_EQUAL(r->one(), 0x50);

    delete r;
}

File: spec/java/src/io/kaitai/struct/spec/TestHelloWorld.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.HelloWorld;
import org.testng.annotations.Test;

import static org.testng.Assert.assertEquals;

public class TestHelloWorld extends CommonSpec {
    @Test
    public void testHelloWorld() throws Exception {
        HelloWorld r = HelloWorld.fromFile(SRC_DIR + "fixed_struct.bin");

        assertEquals(r.one(), 0x50);
    }

}
